### ISSUE TITLE
TypeError in `sampler.emcee`

### ISSUE THREAD

**Issue by Bhavesh012** [phoebe-project/phoebe2#378]
## Error
While running `emcee.sampler` in `run_solver`, `TypeError: '<' not supported between instances of 'str' and 'int'` is thrown out.

## Cause
The error was caused by the version of emcee, "3.0rc1". When the code runs `LooseVersion(emcee.__version__) < LooseVersion("3.0.0")`, it further runs `LooseVersion().version` commands in `python.disutils.version` for comparing. `LooseVersion(emcee.__version).version` gives `[3, 0, 'rc', 1]` which is compared with `[3, 0, 0]`, hence gives the error.

Upgrading `emcee` to `3.0.2` solves the problem. But in future, version like this "3.0rc1" can cause TypeError.

## Traceback

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-48-1de027f35812> in <module>
      5             overwrite=True)
      6 
----> 7 b.run_solver('emcee_solver', niters=300, nwalkers=16, log_level='warning', solution='emcee_sol', progress=True, overwrite=True)
      8 get_ipython().run_line_magic('telepyth', "'emcee solver running complete'")

/usr/local/lib/python3.8/dist-packages/phoebe/parameters/parameters.py in _send_if_client(self, *args, **kwargs)
    398 
    399         else:
--> 400             return fctn(self, *args, **kwargs)
    401 
    402     return _send_if_client

/usr/local/lib/python3.8/dist-packages/phoebe/frontend/bundle.py in run_solver(self, solver, solution, detach, return_changes, **kwargs)
  11084 
  11085         solver_class = getattr(_solverbackends, '{}Backend'.format(solver_ps.kind.title()))
> 11086         params = solver_class().run(self, solver_ps.solver, compute, solution=solution, **{k:v for k,v in kwargs.items() if k not in ['compute']})
  11087         metawargs = {'context': 'solution',
  11088                      'solver': solver_ps.solver,

/usr/local/lib/python3.8/dist-packages/phoebe/solverbackends/solverbackends.py in run(self, b, solver, compute, **kwargs)
    559         if within mpirun, workers should call _run_worker instead of run
    560         """
--> 561         self.run_checks(b, solver,  compute, **kwargs)
    562 
    563         logger.debug("rank:{}/{} calling get_packet_and_solution".format(mpi.myrank, mpi.nprocs))

/usr/local/lib/python3.8/dist-packages/phoebe/solverbackends/solverbackends.py in run_checks(self, b, solver, compute, **kwargs)
   1165             raise ImportError("could not import emcee")
   1166 
-> 1167         if LooseVersion(emcee.__version__) < LooseVersion("3.0.0"):
   1168             raise ImportError("emcee backend requires emcee 3.0+, {} found".format(emcee.__version__))
   1169 

/usr/lib/python3.8/distutils/version.py in __lt__(self, other)
     50 
     51     def __lt__(self, other):
---> 52         c = self._cmp(other)
     53         if c is NotImplemented:
     54             return c

/usr/lib/python3.8/distutils/version.py in _cmp(self, other)
    335         if self.version == other.version:
    336             return 0
--> 337         if self.version < other.version:
    338             return -1
    339         if self.version > other.version:

TypeError: '<' not supported between instances of 'str' and 'int'

```

**Comment by kecnry** [comment]
This is an interesting case.  If emcee used 3.0.0rc1 instead of 3.0rc1, then LooseVersion wouldn't have a problem, but that is obviously outside our control.  I'm considering just catching this TypeError and raising the same ImportError as if the version were insufficient (something like `"ImportError: emcee backend requires a stable release of emcee 3.0+, 3.0rc1 found"`).  

Does that seem like a reasonable solution?  I see you also opened but then closed a PR on this same issue (#377) - was that solution insufficient?

**Comment by Bhavesh012** [comment]
Yes, I think it seems reasonable. We may also suggest the user to upgrade or downgrade to version like, ` Please upgrade to 3.0.2 or downgrade to 3.0.1.`

Only mentioning `3.0rc1` may confuse a user.

And regarding the PR I raised, that was kinda silly. I thought that one version is going `str` and other going `int`. So, thought of converting both in same date type. Later I realised that it's not going to work because the main issue was the way comparison was getting implemented in `disutils.version._cmp()`. So closed the PU cause it wasn't going to help anyway.
